robot_calibration/include/robot_calibration/capture_features.h

```diff
@@ -0,0 +1,124 @@
+#ifndef ROBOT_CALIBRATION_CAPTURE_FEATURES_H
+#define ROBOT_CALIBRATION_CAPTURE_FEATURES_H
+
+
+/**
+* @brief      Verifies the required parameters have been set and, if not, set default values.
+*
+* @param[in]  node  node handle reference
+*
+* @return     bool indicating whether the parameters are all set
+*/
+bool check_parameters(ros::NodeHandle& node);
+
+
+/**
+* @brief      Creates a string for the absoulte path for a file in the home directory.
+*
+* @param[in]  local_dir     name of the local directory.
+*
+* @return     string representing the absolute path to the file
+*/
+std::string get_absolute_directory(std::string local_dir);
+
+
+/**
+* @brief      Creates a map of the available feature finders
+*
+* @param[in]  nh        node handle reference.
+* @param[in]  finders   the location to store the map of feature finders
+*
+* @return     bool indicating whether the feature finders were loaded 
+*/
+bool get_feature_finders(ros::NodeHandle& nh, robot_calibration::FeatureFinderMap& finders);
+
+
+/**
+* @brief      Executes the steps required to automatically capture the feature from a bag of predetermined pose.
+*             This includes commanding the robot to the specified pose then capturing the feature. 
+*
+* @param[in]  nh                node handle reference.
+* @param[in]  chain_manager     object that handles the kinematic chains
+* @param[in]  finders           map of available feature finders
+* @param[in]  feature           feature to use for calibration
+* @param[in]  bag               rosbag object for storing calibration data
+*
+* @return     bool indicating whether the run was successful 
+*/
+bool run_automatic_capture(ros::NodeHandle& nh, 
+                           robot_calibration::ChainManager* chain_manager, 
+                           robot_calibration::FeatureFinderMap& finders,
+                           std::string& feature,
+                           rosbag::Bag& bag);
```

Comment:
Are all of these actually input parameters?

Why did you use a raw pointer for robot_calibration::ChainManager and references for all the other inputs?  Can any of these be made const references instead of just references?